### DISCUSSION TITLE
Why is Link preferred over navigate ?

### DISCUSSION THREAD

**Discussion by gowthamvbhat** [remix-run/react-router#8588] [Q&A]
Hi,
I came across this in the [docs](https://reactrouter.com/docs/en/v6/getting-started/concepts#navigate-function),

> You should have a good reason to use navigate instead of `<Link>`. This makes us very sad:
>
> `<li onClick={() => navigate("/somewhere")} />`
>
> Aside from links and forms, very few interactions should change the URL because it introduces complexity around accessibility and user expectations.

Can someone please elaborate on this ?
Is this performance related or are there any potential problems etc.. ?

Thanks in advance!

**Comment by mfpopa** [comment]
- `Link` is declarative and renders as an `a` tag (better for accessibility), `navigate` is imperative.
- `useNavigate` create waste renders, #7634
